lab_2/main.py

```diff
@@ -89,7 +180,30 @@ def predict_language_knn(unknown_text_vector: list, known_text_vectors: list,
     :param k: the number of neighbors to choose label from
     :param metric: specific metric to use while calculating distance
     """
-    pass
+    if not (
+        isinstance(unknown_text_vector, list)
+        and all(isinstance(n, (int, float)) for n in unknown_text_vector)
+        and isinstance(known_text_vectors, list)
+        and all(isinstance(m, list) for m in known_text_vectors)
+        and isinstance(language_labels, list)
+        and all(isinstance(s, str) for s in language_labels)
+        and len(known_text_vectors) == len(language_labels)
+        ):
+        return None
+    if metric == 'euclid':
+        calc_dist = calculate_distance
+    elif metric == 'manhattan':
+        calc_dist = calculate_distance_manhattan
+    distances = [calc_dist(unknown_text_vector, knw_vector) for knw_vector in known_text_vectors]
+    sorted_distances = (sorted(distances))[:k]
+    langs = [language_labels[distances.index(dist)] for dist in sorted_distances]
+    langs = sorted(zip(langs, sorted_distances), key=lambda x: x[1])
+    lang_count = {}
+    for lang, _ in langs:
```

Comment:
can you explain what `_` is?

Reply:
_ is used to ignore a value of specific location in the tuple.
Langs is a list of tuples, e.g. lang = [('en', 0.57297), ('de', 0.60025), ('de', 0.67904)].  I don't need to use the values of distances (0.57297, 0.60025, 0.67904) in the next part of my code (only values of languages 'en', 'de', 'de' will be used further). I use _, because the second element in all tuples in the list is unnecessary for me.